plugin/meshnet.go

```diff
@@ -75,20 +76,20 @@ func loadConf(bytes []byte) (*netConf, *current.Result, error) {
 	return n, result, nil
 }
 
-// getVxlanSource uses netlink to query the IP and LinkName of the interface with default route.
-func getVxlanSource() (string, string, error) {
-	log.Infof("Looking up a default route to get the intf and IP for vxlan")
-	r, err := netlink.RouteGet(net.IPv4(1, 1, 1, 1))
-	if (err != nil) && len(r) < 1 {
-		return "", "", fmt.Errorf("failed to get default route: %s\n%+v", err, r)
-	}
-	srcIP := r[0].Src.String()
-	link, err := netlink.LinkByIndex(r[0].LinkIndex)
-	if err != nil {
-		return "", "", fmt.Errorf("failed to get link by its index: %s", err)
+// getVxlanSource uses netlink to get the iface reliably given an IP address.
+func getVxlanSource(nodeIP string) (string, string, error) {
+
+	ifaces, _ := net.Interfaces()
+	for _, i := range ifaces {
+		addrs, _ := i.Addrs()
+		for _, a := range addrs {
+			if strings.Contains(a.String(), nodeIP) {
```

Comment:
why do you use strings.Contains and not `a.String() != nodeIP`?  is it because a.String() can contain port number?

Reply:
`net.Addr.String()` only documents that it might contain a port, but in my testing it returns a CIDR, e.g., "10.20.30.40/31".

E.g.:
```
mimesis > cat main.go
package main

import (
	"log"
	"net"
)

func main() {
	ifaces, _ := net.Interfaces()
	for _, i := range ifaces {
		addrs, _ := i.Addrs()
		for _, a := range addrs {
			log.Printf("Iface: %v, Addr: %v", i, a.String())
		}
	}
}

~/Documents/projects/fizzy/mimesis/scratch/test master*
mimesis > go run main.go
2022/03/10 09:26:14 Iface: {1 16384 lo0  up|loopback|multicast}, Addr: 127.0.0.1/8
2022/03/10 09:26:14 Iface: {1 16384 lo0  up|loopback|multicast}, Addr: ::1/128
2022/03/10 09:26:14 Iface: {1 16384 lo0  up|loopback|multicast}, Addr: fe80::1/64
2022/03/10 09:26:14 Iface: {4 1500 en6 ac:de:48:00:11:22 up|broadcast|multicast}, Addr: fe80::aede:48ff:fe00:1122/64
2022/03/10 09:26:14 Iface: {6 1500 en0 88:66:5a:1e:2f:d3 up|broadcast|multicast}, Addr: fe80::847:663d:aee5:ea09/64
2022/03/10 09:26:14 Iface: {6 1500 en0 88:66:5a:1e:2f:d3 up|broadcast|multicast}, Addr: 192.168.86.45/24
```

Reply:
gotcha.